rust/src/lib.rs

```diff
@@ -266,21 +267,30 @@ impl TransactionBody {
         self.metadata_hash.clone()
     }
 
+    pub fn set_validity_start_interval(&mut self, validity_start_interval: Slot) {
+        self.validity_start_interval = Some(validity_start_interval)
+    }
+
+    pub fn validity_start_interval(&self) -> Option<Slot> {
+        self.validity_start_interval
+    }
+
     pub fn new(
         inputs: &TransactionInputs,
         outputs: &TransactionOutputs,
         fee: &Coin,
-        ttl: u32,
+        ttl: Slot,
```

Comment:
I left this field as required in the constructor since:
1) you mentioned it's a good idea for users to put it in
2) the API doesn't change

I can take it out if you want

Reply:
Yeah personally I can't think of a reason why somebody would ever not want to specify a TTL. However, it may be better to make it optional to properly mirror the spec. I'm okay with either options